app/code/Magento/Downloadable/Controller/Download/Sample.php

```diff
@@ -46,7 +46,9 @@ public function execute()
         $sampleId = $this->getRequest()->getParam('sample_id', 0);
         /** @var \Magento\Downloadable\Model\Sample $sample */
         $sample = $this->_objectManager->create(\Magento\Downloadable\Model\Sample::class)->load($sampleId);
```

Comment:
Could you also add `\Magento\Downloadable\Model\SampleFactory` in constructor and create `$sample` object from it, not via ObjectManager?

Reply:
Done